library/src/main/java/com/northconcepts/templatemaster/form/editor/NumberEditor.java

```diff
@@ -0,0 +1,42 @@
+package com.northconcepts.templatemaster.form.editor;
+
+import java.text.DecimalFormat;
+
+import com.northconcepts.templatemaster.form.CrudResource;
+import com.northconcepts.templatemaster.form.FieldDef;
+import com.northconcepts.templatemaster.form.presenter.NumberPresenter;
+
+public class NumberEditor implements FieldValueEditor {
+
+    public static final NumberPresenter DOLLARS_NO_CENTS = new NumberPresenter("$#,##0");
+    public static final NumberPresenter DOLLARS_CENTS = new NumberPresenter("$#,##0.00");
+    public static final NumberPresenter WHOLE_NO_GROUPING = new NumberPresenter("0");
+    public static final NumberPresenter WHOLE_GROUPING = new NumberPresenter("#,##0");
+    public static final NumberPresenter DECIMAL_NO_GROUPING = new NumberPresenter("0.#");
+    public static final NumberPresenter DECIMAL_GROUPING = new NumberPresenter("#,##0.#");
+
+    private DecimalFormat decimalFormat;
```

Comment:
Let's make the field final.
```suggestion
    private final DecimalFormat decimalFormat;
```

Reply:
Done.